src/com/example/Demo_Subj/Subject.java

```diff
@@ -258,4 +208,77 @@ private Bitmap mirrorBitmap(Bitmap b){
         b = Bitmap.createBitmap(b, 0, 0, b.getWidth(), b.getHeight(), matrixMirror, false);
         return b;
     }
+
+    public void tick(){
+        //Animation für das Subjekt, übernommen aus Subjekt
+        /*****************************************************************************************/
+        intel.tick();                   //KI aufrufen
+        //hier muss später noch ein Aufruf zum ändern der Animation eingefügt werden, z.B. durch die KI oder durch RoomActivity selbst
+        /*switch (animation) {
+            case 0:     calculatePosition();
+                break;
+
+            //Weiter Switch-Cases für weitere Animationen
+
+            default:    aktBitmap = subjStandBitmap;        //entweder das oder eine Fehlermeldung
+                break;
+        }*/
+        calculatePosition();
+    }
+
+    public Bitmap getBitmap(){
```

Comment:
Jetzt sind ja die Bitmaps doch wieder hier gelagert? Ich weiß nicht, ob du nicht verstanden hast wie das laufen sollte: Subjekt enthält wieder Listen von Bitmap-IDs (Stehen, Laufen, von mir aus auch noch Laufen in die andere Richtung wenn es dafür eigene Bilder geben sollte), und gibt an den Renderer nur eine Zustands-Information "Ich laufe gerade" oder "Ich stehe gerade". Der Renderer würde in seiner Initialisierung die Bilder die zu den IDs gehören bei sich laden und dann zu jedem Tick gucken, "soll ich jetzt animieren (Bilder durchschalten), weil es läuft, oder soll ich das Standbild anzeigen"? Jetzt ist das hier ja quasi alles wie vorher... Oder übersehe ich da etwas?

Nochmal konkret: Das Subjekt kennt nur die Nummern seiner Bilder (R.drawable.etc) und seinen Zustand (Position X, Y sowie "laufend" oder "stehend") - all diese Sachen kann sich der Renderer holen über Methoden (getPictureID, getState, getXpos oder ähnlich) und ER verwaltet dann intern das Laden der Bilder und das Weiterklickern der Animation.